spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/logging/LoggersGroupEndpointAutoConfigurationTests.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.logging;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.boot.actuate.logging.LoggerGroupsEndpoint;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.logging.LoggerGroups;
+import org.springframework.boot.logging.LoggingSystem;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests for {@link LoggersGroupEndpointAutoConfiguration}.
+ *
+ * @author HaiTao Zhang
+ */
+public class LoggersGroupEndpointAutoConfigurationTests {
+
+	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+			.withConfiguration(AutoConfigurations.of(LoggersGroupEndpointAutoConfiguration.class))
+			.withUserConfiguration(LoggingConfiguration.class);
+
+	@Test
+	void runShouldHaveEndpointBean() {
+		this.contextRunner.withPropertyValues("management.endpoints.web.exposure.include=loggergroups")
+				.run((context) -> assertThat(context).hasSingleBean(LoggerGroupsEndpoint.class));
+	}
+
+	@Test
+	void runWithLoggerGroupsEndpointDisabledShouldNotHaveEndpointBean() {
+		this.contextRunner.withPropertyValues("management.endpoint.loggergroups.enabled=false")
+				.run((context) -> assertThat(context).doesNotHaveBean(LoggerGroupsEndpoint.class));
+	}
+
+	@Test
+	void runWithEndpointNotExposedShouldNotHaveEndpointBean() {
+		this.contextRunner.run((context) -> assertThat(context).doesNotHaveBean(LoggerGroupsEndpoint.class));
+	}
+
+	@Test
+	void runWithNoLoggingSystemShouldNotHaveEndpointBean() {
+		this.contextRunner.withPropertyValues("org.springframework.boot.logging.LoggingSystem=none")
+				.run((context) -> assertThat(context).doesNotHaveBean(LoggerGroupsEndpoint.class));
+	}
+
+	@Test
+	void runWithNoLoggerGroupsShouldNotHaveEndpointBean() {
```

Comment:
I am not entirely sure I get the rationale of this. IMO, If we go down the route of having a dedicated endpoint, I don't think it should 404 depending of the fact that at least a group is defined or not.